src/eckit/types/Time.cc

```diff
@@ -8,34 +8,64 @@
  * does it submit to any jurisdiction.
  */
 
-#include <regex>
+#include <cctype>
 #include <cmath>
+#include <cstddef>
+#include <regex>
+#include <string>
+#include <string_view>
 
-#include "eckit/eckit.h"
-
+#include "eckit/exception/Exceptions.h"
 #include "eckit/persist/DumpLoad.h"
 #include "eckit/types/Time.h"
 #include "eckit/utils/Hash.h"
-#include "eckit/utils/Tokenizer.h"
 
 namespace {
-    static thread_local std::regex digits_time_("^-?[0-9]+$");
-    static thread_local std::regex float_hours_("^-?[0-9]*\\.[0-9]+$");
-    static thread_local std::regex hhmmss_("^([0-9]+):([0-5]?[0-9])(:[0-5]?[0-9])?$");
-    static thread_local std::regex ddhhmmss_("^-?([0-9]+[dD])?([0-9]+[hH])?([0-9]+[mM])?([0-9]+[sS])?$");
-}
-
-namespace eckit {
 
 //----------------------------------------------------------------------------------------------------------------------
 
-inline void printTime(std::ostream& s, long n) {
+const std::regex hhmmss_("^([0-9]+):([0-5]?[0-9])(:[0-5]?[0-9])?$");
+const std::regex ddhhmmss_("^-?([0-9]+[dD])?([0-9]+[hH])?([0-9]+[mM])?([0-9]+[sS])?$");
+
+// DIGITS: "^-?[0-9]+$"
+// FLOAT:  "^-?[0-9]*\\.[0-9]+$"
+enum class TimeFormat { UNKOWN, OTHER, DIGITS, DECIMAL };
```

Comment:
`UNKNOWN` is treated as erroneous input, it go stronger in the name and call it `INVALID`